modules/oe_translation_cdt/modules/oe_translation_cdt_mock/src/Plugin/ServiceMock/ServiceMockBase.php

```diff
@@ -149,9 +152,10 @@ public function getResponseFromFile(string $filename): string {
    * {@inheritdoc}
    */
   public function applies(RequestInterface $request, array $options): bool {
-    // Replace all URL parameters with regex wildcards.
-    $endpoint_pattern = preg_replace('/:\w+/', '[^/]*', $this->getEndpointUrl());
-    return preg_match("|^$endpoint_pattern$|", (string) $request->getUri()) === 1;
+    // Replace all path parameters with regex wildcards.
+    $path_pattern = preg_replace('/:\w+/', '[^/]*', $this->getEndpointUrlPath());
+    $url_pattern = rtrim(Settings::get('cdt.base_api_url'), '/') . $path_pattern;
```

Comment:
Doing this string concat is not perfect (it could be done and returned by CDT itself), but I left it here as a lesser evil. All the other ideas lead to significant changes inside the CDT.